packages/suspense-website/src/examples/createCache/cacheWithLRU.ts

```diff
@@ -0,0 +1,19 @@
+import { createCache, Record } from "suspense";
+
+const load = async () => null as any;
+
+// REMOVE_BEFORE
+
+import LRUCache from "lru-cache";
+
+createCache<[userId: string], JSON>({
+  config: {
+    getCache: (onEvict) =>
+      new LRUCache<string, Record<JSON>>({
+        max: 100,
+        dispose: (value, key) => onEvict(key),
```

Comment:
> Secondly, I believe dispose is called _before_ it's actually removed from the cache. The docs say that it's removed in the next tick, so something like `setTimeout(() => onEvict(key), 1)` works as well

This doesn't seem necessary. There's no harm in evicting a frame earlier than the cache, is there?

Reply:
The problem is that if we notify a frame earlier, then our subscribers will receive the unevicted status rather than the post eviction status

Reply:
Why?

Forgetting about the details of this LRU for a second– calling `evict` on a cache will _synchronously evict_ the record and the status will then be "not-found" for any subscribers. Why would the LRU's internal details change this?

(Sorry if I'm overlooking something...it's been a long day.)

Reply:
I'm not sure why it's implemented this way but since dispose is called before it's actually evicted, calling the `onEvict` callback synchronously means our subscribers will notify of this particular lru cache _pre_ eviction, since the actual eviction will happen in the next frame. Meaning when it calls getStatus, it will appear to still be whatever it was pre eviction. 

Sorry about the long day, hope you get some rest

Reply:
To further clarify, the way we synchronize with the cache provided by `getCache` is by providing an `onEvict` callback to the `getCache` function.

Reply:
> calling the `onEvict` callback synchronously means our subscribers will notify of this particular lru cache _pre_ eviction, since the actual eviction will happen in the next frame. Meaning when it calls getStatus, it will appear to still be whatever it was pre eviction. 

Ah. I think I understand. I have not dug into the implementation of how the `onEviction` callback worked. I assumed it was calling `evict` but it's actually calling a different method `onExternalCacheEviction`. Any reason for that? Were you worried that calling `evict` would be unnecessarily wasteful?

If we _did_ call `evict`– then timing sensitivity issues like this could be avoided.

Reply:
Yeah, my reasoning was that these cache implementations would evict themselves when they needed to, so our task was just to keep up to date with the work done. Generally this isn't an issue when the onEvict is called after the eviction has occurred, but with this particular implementation it is a bit clunky

Reply:
We could just call evict ourselves, and if the cache had already evicted when `onEvict` was called, then it's just a noop. Not sure if that would cause issues with this particular lru-cache implementation. Only reason I included it in the docs was because it's the most popular implementation i know of in the ecosystem

Reply:
> my reasoning was that these cache implementations would evict themselves when they needed to, so our task was just to keep up to date with the work done

Agreed, but I think little "foot guns" like forgetting a `setTimeout` are things that would be nice to avoid if possible :🤪

Reply:
FWIW I updated our unit tests to use `lru-cache`. I'll add a test later this evening for the edge case we're talking about. Hopefully I'll have the time and mental energy to dig into this stuff more fully this weekend.

Reply:
Awesome! I'm not sure if it's still issue now that they've released a new major. 

Have a great weekend :D happy sabbath 🙇